### PR TITLE
chore: Bump Buf to v1.36.0

### PR DESCRIPTION
**Pull request by codingllama**
Update to the latest minor. No codegen or config changes.

* https://github.com/bufbuild/buf/releases/tag/v1.36.0

### CODE REVIEW

**Comment by codingllama** [comment]
I'll likely hold this until it lands on Homebrew, plus a bit more in case v1.36.1 follows it.

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**
